### ISSUE TITLE
[v5] Data migration from v4.3 to v5.0 fails with dependency errors

### ISSUE THREAD

**Issue by loadtheaccumulator** [reportportal/reportportal#1102]
**Describe the bug**
When migrating data from ReportPortal v4.3 to v5.0 via the provided docker-compose, the process fails with multiple dependency errors.

**To Reproduce**
Install new 5.0 system or clean up per the instructions and run the process again.
Reproducible every time.

**Expected behavior**
I expected data to be moved from the mongo database to the postgres database without issue.

**Screenshots**
[reportportal_datamigration_dependencyissue_debug.txt](https://github.com/reportportal/reportportal/files/5028325/reportportal_datamigration_dependencyissue_debug.txt)

**Desktop (please complete the following information):**
 - OS: N/A
 - Browser: N/A
 - Version: N/A

**Smartphone (please complete the following information):**
 - Device: N/A
 - OS: N/A
 - Browser: N/A
 - Version: N/A

**Additional context**
Settings migration ran successfully per the instructions.
The attached log file was run with DEBUG=1.
Operating systems are CentOS and RHEL.
My migration systems are still running from the point of failure, so please let me know if there is additional data you need to resolve this issue.

Cheers!

**Comment by evjlobanova** [comment]
@pbortnik could you please take a look?

**Comment by pbortnik** [comment]
Hi @loadtheaccumulator,
Can you execute the next command according to the MongoDB database. Do you have any response?
`db.getCollection('testItem').find({"start_time" : {$exists : false}})`

**Comment by loadtheaccumulator** [comment]
@pbortnik I went through that step per another ticket I had run across, deleted the entry, and cleaned up before restarting the process.
It still shows nothing in response...
> db.getCollection('testItem').find({"start_time" : {$exists : false}})
>

**Comment by DaryaZhukovaTR** [comment]
Hello, I'm seeing the similar issue during data migration:

07-08-2020 17:04:05.743 [main] ERROR org.springframework.boot.SpringApplication.reportFailure - Application startup failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'jobsConfiguration': Unsatisfied dependency expressed through field 'levelItemsFlow'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'levelItemsFlow' defined in class path resource [com/epam/reportportal/migration/steps/items/ItemsStepConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [java.util.List]: Factory method 'levelItemsFlow' threw exception; nested exception is java.lang.NullPointerException

No response:
> db.getCollection('testItem').find({"start_time" : {$exists : false}})

**Comment by loadtheaccumulator** [comment]
Hi @pbortnik and @evjlobanova We are blocked on this one and can't move forward with ReportPortal v5.
Are there additional debug steps we can take on our end to help?
Thanks,
Jonathan

**Comment by DaryaZhukovaTR** [comment]
Dropping mongo collections: 'filterMapping', 'widgetMapping', 'optimizeTest' helped to resolve the issue I faced.
Thanks!

**Comment by loadtheaccumulator** [comment]
> Dropping mongo collections: 'filterMapping', 'widgetMapping', 'optimizeTest' helped to resolve the issue I faced.
> Thanks!

That didn't work for us.

**Comment by pbortnik** [comment]
Hi @loadtheaccumulator,
Need to ask you some more questions.
1. Do you have any documents in testItem collection?
2. Do you have any documents in optimizeTest collection after data-migration running?
3. Do you have the latest version of data-migration image (check it with docker-compose -f datamigration_filename pull)

**Comment by loadtheaccumulator** [comment]
> * Do you have any documents in testItem collection?
> 
Using "db.getCollection('testItem').find()"
yes. this command returns data

> * Do you have any documents in optimizeTest collection after data-migration running?
> 
Using "db.getCollection('optimizeTest').find()"
yes. this command returns data

> * Do you have the latest version of data-migration image (check it with docker-compose -f datamigration_filename pull)
There is now one dated 9 days ago. The latest image was from 2 months ago when we experience the issue.

The migration attempt after the pull appeared to get further than the previous runs, but still error'd.
(see attached file)
[reportportal_datamigration_dependencyissue_debug_20200819.log.txt](https://github.com/reportportal/reportportal/files/5100589/reportportal_datamigration_dependencyissue_debug_20200819.log.txt)

**Comment by loadtheaccumulator** [comment]
@evjlobanova @pbortnik We're stuck and can't move to Version 5. Any guidance you can provide on how we need to proceed would be greatly appreciated.

**Comment by pbortnik** [comment]
Hi @loadtheaccumulator,
It's a very interesting issue, and i need a kind of your MongoDB database dump to reproduce and investigate the issue. Is it possible for you ?

**Comment by loadtheaccumulator** [comment]
> Hi @loadtheaccumulator,
> It's a very interesting issue, and i need a kind of your MongoDB database dump to reproduce and investigate the issue. Is it possible for you ?

Hey @pbortnik I am still looking into whether this is possible.

**Comment by loadtheaccumulator** [comment]
@pbortnik Since the latest errors point to line 127 and start_time, we focused our latest look on date ranges.
We are able to migrate data from 2019 through 2020. Pre-2019 data is failing. Possible problem stemming from the ReportPortal version? I'll have to see if we can determine what specific version was in use then.

(and I'm still looking into whether we can share the database itself).

**Comment by pbortnik** [comment]
No, it is not rp version.
It can be database dump pre-2019 data if you are sure that problem somewhere there.

**Comment by loadtheaccumulator** [comment]
Hey Pavel, We're pretty consistent on the pre-2019 data, so I'm going to close this and I'll re-open or open a new ticket if anyone needs data from 2018. Thanks, Jonathan
